### PR TITLE
Impala : Add support for trigonometric functions (sin, cos, tan)

### PR DESCRIPTION
**Pull request by deepkset**
Issue #2142: Impala : Add support for trigonometric functions (sin, cos, tan)

### CODE REVIEW

**Comment by jreback** [comment]
pls add a release note and referene the original issue https://github.com/ibis-project/ibis/issues/2141

also, did this change an xfail test to a passing one?

**Comment by datapythonista** [comment]
@deepkset still interested in getting this merged? If so, please address the comments and merge master.

File: ibis/impala/compiler.py [hunk 1]
```diff
@@ -1165,6 +1165,9 @@ def _sign(translator, expr):
     ops.Log10: unary('log10'),
     ops.DecimalPrecision: unary('precision'),
     ops.DecimalScale: unary('scale'),
+    ops.Sin: unary('sin'),
```
Comment (jreback) [inline ibis/impala/compiler.py:1168]: these are currently xfailed in the tests I *think* can you confirm

**Comment by datapythonista** [comment]
Closing as stale. @deepkset if you have interest in continuing this work, please let me know.

**Pull request closed without merging**